packages/clerk-js/src/core/clerk.ts

```diff
@@ -926,6 +926,18 @@ export class Clerk implements ClerkInterface {
       eventBus.dispatch(events.TokenUpdate, { token: null });
     }
 
+    const hasSessionTaskToResolve = !!newSession?.tasks?.length;
+    if (hasSessionTaskToResolve && this.#options.standardBrowser) {
+      const hasEventHandler = eventBus.has(events.InternalComponentNavigate);
+      if (hasEventHandler && newSession) {
+        await new Promise<void>(resolveNavigation =>
+          eventBus.dispatch(events.InternalComponentNavigate, { resolveNavigation, session: newSession }),
+        );
+      } else {
+        await this.navigate(newSession?.currentTask?.__internal_getAbsoluteUrl(this.#options, this.environment));
```

Comment:
Idea: comment explaining why we land in each of these branches (custom flow vs our components)